src/suite_starter/suite_starter.py

```diff
@@ -97,31 +115,37 @@ def suite_runner_callback(self, event, _):
         :return: Whether event was ACK:ed or not.
         :rtype: bool
         """
-        suite_id = event.meta.event_id
-        FORMAT_CONFIG.identifier = suite_id
-        LOGGER.info("Received a TERCC event. Build data for ESR.")
-        data = {"EiffelTestExecutionRecipeCollectionCreatedEvent": json.dumps(event.json)}
-        data["suite_id"] = suite_id
-
-        job = Job(in_cluster=bool(os.getenv("DOCKER_CONTEXT")))
-        job_name = job.uniqueify(f"suite-runner-{suite_id}").lower()
-        data["job_name"] = job_name
-
-        LOGGER.info("Dynamic data: %r", data)
-        LOGGER.info("Static data: %r", self.etos.config.get("configuration"))
-        try:
-            assert data["EiffelTestExecutionRecipeCollectionCreatedEvent"]
-        except AssertionError as exception:
-            LOGGER.critical("Incomplete data for ESR. %r", exception)
-            raise
-
-        body = job.load_yaml(
-            self.suite_runner_template.format(**data, **self.etos.config.get("configuration"))
-        )
-        LOGGER.info("Starting new executor: %r", job_name)
-        job.create_job(body)
-        LOGGER.info("ESR successfully launched.")
-        return True
+        with self.tracer.start_as_current_span("suite", context=context.get_current()) as span:
+            suite_id = event.meta.event_id
+            FORMAT_CONFIG.identifier = suite_id
+            LOGGER.info("Received a TERCC event. Build data for ESR.")
+            data = {"EiffelTestExecutionRecipeCollectionCreatedEvent": json.dumps(event.json)}
+            data["suite_id"] = suite_id
+            data["otel_context"] = self._get_current_context()
```

Comment:
What happens if there is no context set? ETOS must work even if it is not

Reply:
Made adjustments so that it works without errors in the log.